From 1995ef6d9459931170f5718221104d7eb8c35eba Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 9 May 2017 16:32:09 -0400
Subject: [PATCH 24/26] efi_generate_file_device_path(): make all error paths
 use "goto err;"

Honestly I'm just trying to shut coverity up about checking
child_devpath for NULL-ness twice on the other two error paths.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/creator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/creator.c b/src/creator.c
index ccd0faf..21fc6b0 100644
--- a/src/creator.c
+++ b/src/creator.c
@@ -339,7 +339,7 @@ efi_generate_file_device_path(uint8_t *buf, ssize_t size,
 	rc = find_file(filepath, &child_devpath, &relpath);
 	if (rc < 0) {
 		efi_error("could not canonicalize fs path");
-		return -1;
+		goto err;
 	}
 
 	rc = find_parent_devpath(child_devpath, &parent_devpath);
-- 
2.12.2

